### ISSUE TITLE
Add all the averaging methods available for Precion, Recall and F-score in sklearn, natively in tensorflow or tensorflow-addons

### ISSUE THREAD

**Issue by naveen-marthala** [tensorflow/tensorflow#52793]
<em>Please make sure that this is a feature request. As per our [GitHub Policy](https://github.com/tensorflow/tensorflow/blob/master/ISSUES.md), we only address code/doc bugs, performance issues, feature requests and build/installation issues on GitHub. tag:feature_template</em>

**System information**
- TensorFlow version (you are using): 2.4.3
- Are you willing to contribute it (Yes/No): Yes, but I have very little knowledge on tensorflow's graph execution works. So, I can't contribute.

**Describe the feature and the current behavior/state.**
Currently the [Recall](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Recall)/[Precision ](https://www.tensorflow.org/api_docs/python/tf/keras/metrics/Precision)metric available in tensorflow only provides binary averaged score; and [F1Score](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/F1Score) and [FBetaScore](https://www.tensorflow.org/addons/api_docs/python/tfa/metrics/FBetaScore). It would be very convenient and avoids the need to write (half-baked, at least in my case) custom metrics from user side, if other averaging methods available in sklearn, that is 'macro', 'micro' and 'samples'(for multi-label problems), are also implemented in either tensorflow or tensorflow-addons.

**Will this change the current api? How?**
I presume, No.

**Who will benefit with this feature?**
every one working on multi-class and multi-label classification problems, who like to use those averaging methods.

**Any Other info.**
All aforementioned metrics from sklearn's API docs are hyperlinked below: 
[sklearn's Recall](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.recall_score.html#sklearn.metrics.recall_score), [sklearn's Precision](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.precision_score.html#sklearn.metrics.precision_score), [sklearn's F1Score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.f1_score.html#sklearn.metrics.f1_score) and [sklearn's FBeta Score](https://scikit-learn.org/stable/modules/generated/sklearn.metrics.fbeta_score.html#sklearn.metrics.fbeta_score)

**Comment by jvishnuvardhan** [comment]
@naveen-marthala This is a feature request for Keras.

Please note that Keras development moved to another repository to focus entirely on only keras. Could you please repost this issue on [keras-team/keras repo.](https://github.com/keras-team/keras/issues). Thanks!

**Comment by naveen-marthala** [comment]
ok @jvishnuvardhan

**Comment by naveen-marthala** [comment]
@jvishnuvardhan, opened it [here in keras repo](https://github.com/keras-team/keras/issues/15808).
